File: valid_palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    // converting all to lower case
    s = s.toLowerCase();

    // new string where only letters and number digits will be included 
    let s_new = '';

    // loop through the original string to get s_new 
    for (let i=0; i< s.length; i++){
        if ( (s[i] >= 'a' && s[i]<='z') || (s[i]>='0' && s[i]<='9') ){
            s_new = s_new + s[i];
        }
    }

    // compare first with last, then second with second last, ... until middle element
    for (let i=0; i <= Math.floor(s_new.length/2); i++){
        if (s_new[i] != s_new[(s_new.length)-1-i]){
            return false
        }
    }
    return true
};

File: valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        # converting all to lower case
        s = s.lower()

        # new string where only letters and number digits will be included 
        s_new = ''

        # loop through the original string to get s_new 
        for x in s:
            if (x >= 'a' and x <= 'z' or x>='0' and x<='9'):
                s_new = s_new + x
        
        # compare first with last, then second with second last, ... until middle element
        for i in range(0,(len(s_new)//2)):
            if (s_new[i] != s_new[(len(s_new)-1-i)]):
                return False
        return True     

        